quinn/src/connection.rs

```diff
@@ -458,6 +459,11 @@ where
         self.0.lock("stats").inner.stats()
     }
 
+    /// Current state of the congestion control algorithm
+    pub fn congestion(&self) -> Box<dyn Controller> {
```

Comment:
```suggestion
    pub fn congestion_state(&self) -> Box<dyn Controller> {
```
How can a user downcast this? Do we really want to expose it, or would it be better off as something you patch in locally when needed for development?

Reply:
I don't think it should just be patched in locally. Hopefully I'm not the only one who will do development on CCAs. As for downcasting, that's not something I've needed to do. I just print out the debug representation.

Reply:
If the intended use is just for diagnostics during development, we should at least document that fact.

> I just print out the debug representation.

How does that work? `Controller` doesn't imply `Debug`.

Reply:
controller does imply debug since controller implements send + debug

Reply:
Oh, right, that was added in this PR. Feels like a bit of a hack, but I guess it's harmless.

Reply:
it's definitely a hack- any better way to achieve printing this data out given a connection is  welcome